apps/site/assets/ts/stop/components/DepartureTimes.tsx

```diff
@@ -0,0 +1,85 @@
+import formatDistanceToNow from "date-fns/formatDistanceToNow";
+import React, { ReactElement } from "react";
+import usePredictionsChannel from "../../hooks/usePredictionsChannel";
+import { isACommuterRailRoute } from "../../models/route";
+import { DirectionId, Route, Stop } from "../../__v3api";
+import renderFa from "../../helpers/render-fa";
+
+interface DepartureTimesProps {
+  route: Route;
+  stop: Stop;
+  directionId: DirectionId;
+}
+
+const formatTime = (timeString: string, isCR: boolean): string => {
+  const d = new Date(timeString);
+  if (isCR) {
+    return d.toLocaleTimeString();
+  }
+  return formatDistanceToNow(d, { addSuffix: true });
+};
+
+/* istanbul ignore next */
+/**
+ * A proof-of-concept component illustrating a usage of the
+ * usePredictionsChannel hook to fetch live predictions for a specific
+ * route/stop/direction, sorted by date.
+ *
+ */
+const DepartureTimes = (
+  props: DepartureTimesProps
+): ReactElement<HTMLElement> => {
+  const { route, stop, directionId } = props;
+  const isCR = isACommuterRailRoute(route);
+  const predictionsByHeadsign = usePredictionsChannel(
+    route.id,
+    stop.id,
+    directionId,
+    isCR ? 1 : 2
+  );
+
+  if (!Object.keys(predictionsByHeadsign).length) {
+    return (
+      <div className="departure-card__headsign">
+        <div className="departure-card__headsign-name">
+          {route.direction_destinations[directionId]}
+        </div>
+        <div>No predictions.</div>
+        <button
+          type="button"
+          aria-label={`Open upcoming departures to ${route.direction_destinations[directionId]}`}
+        >
+          {renderFa("", "fa-angle-right")}
+        </button>
+      </div>
+    );
+  }
+  return (
+    <>
+      {Object.entries(predictionsByHeadsign).map(([headsign, predictions]) => {
+        return (
+          <div key={headsign} className="departure-card__headsign">
```

Comment:
suggestion (non-blocking): This also feels like something that could be refactored out into a named, sub-component function.